### PR TITLE
Images manage on tournament module

### PR DESCRIPTION
**Pull request by taltaub22**
Everything here is working with base64 encoding
That means that images are retrieved and sent as text.

To save an image:
POST /image/
With body of: 
{imageName: name, image: base64encoding}

### CODE REVIEW

File: routers/imagesRouter.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+'use strict'
+const express = require('express')
+const MongoClient = require('mongodb').MongoClient
+const MsLogger = require('@first-lego-league/ms-logger').Logger()
+const {authroizationMiddlware} = require('@first-lego-league/ms-auth')
```
Comment (idanstark42) [inline routers/imagesRouter.js]: Notice which of these you use and which not

File: routers/imagesRouter.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+const adminAction = authroizationMiddlware(['admin', 'development'])
+
+const router = express.Router()
+
+router.get('/all', (req, res) => {
+  res.send(getAllImages())
+})
+
+router.get('/:imageName', (req, res) => {
+  res.send(getImage(req.params.imageName))
```
Comment (idanstark42) [inline routers/imagesRouter.js]: If you throw an error for an image that doesn't exist, you sold catch it

File: logic/images.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+function getImage (name) {
+  const image = getAllImagesNames().find(x => x == name)
+  if (!image) {
+    MsLogger.error(`trying to get image ${name} but it does not exists`)
+    throw 'Image does not exists'
+  }
+
+  return {
+    'name': image,
+    'image': base64Sync(path.resolve(IMAGES_DIR, image))
```
Comment (idanstark42) [inline logic/images.js]: This lines and lines 29-30 are voice repetition. You can export them to a single function

File: routers/imagesRouter.js (hunk 2 above)
Comment (taltaub22) [reply routers/imagesRouter.js]: Not here, beacuse its causes 500 and sends the error to the user. Express handles it for me

File: logic/images.js (hunk 3 above)
Comment (taltaub22) [reply logic/images.js]: Do I really? The library is doing all the work and wraps everything to one single function that get the path.

Comment (idanstark42) [reply logic/images.js]: I meant the two lines that make it into an object with `name` and `image` properties. This object apears in a couple of places, and it generated exactly the same

File: routers/imagesRouter.js (hunk 2 above)
Comment (idanstark42) [reply routers/imagesRouter.js]: It's kind of a bad practice... The client should get a huma-readable error

**Comment by taltaub22** [comment]
I think because we are talking with the file system its better to use sync functions

**Comment by idanstark42** [comment]
Is this just a feeling or can you back this up? Because IO operations are dangerous, and may take time, and so they probably should be async in my opinion

**Comment by taltaub22** [comment]
Just a feeling, And then I read this article https://nodejs.org/en/docs/guides/blocking-vs-non-blocking/
I forgot that JS is single threaded and therefore sync operations are blocking **everything**

**Comment by taltaub22** [comment]
@idanstark42

**Comment by idanstark42** [comment]
You're still using the `sync` methods, just wrapping them in promises. This is a good solution if [promisify](https://www.npmjs.com/package/promisify) didn't exist. You should `promisify` the `async` functions, i.e. `base64Sync` turn into `promisify(base64)` and then you can use them as promises. Creating you own promise is almost never required.

**Pull request closed without merging**